### PR TITLE
Adds Random Phobia as a Negative Trait

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

what it says on the tin

## Why It's Good For The Game

idk why this hasnt been implemented yet, seems perfect to me

## Changelog
:cl:
add: a new negative trait, random phobia
/:cl:

also should i remove the fear of the dark one or consolidate it or what?

### CODE REVIEW

**Review by nemvar — CHANGES_REQUESTED** [review]
Doesn't remove phobia on remove.

**Comment by wesoda25** [comment]
> Doesn't remove phobia on remove.

wdym

**Comment by Time-Green** [comment]
There’s no code to remove the phobia if the trait is somehow removed

**Comment by wesoda25** [comment]
It shouldn't be.... Whats the point of it? From what I can tell most traits don't have code for removal, unless its handled elsewhere.

**Comment by nemvar** [comment]
There are admin options to add/remove quirks. There are some quirks that support it, others that don't. I see value in having that functionality, so I would add it.

**Comment by anconfuzedrock** [comment]
I'd complain "what if a sec officer gets commandphobia or a lizard is scared of snakes" but then I remembered we have psicodine if that works

**Comment by wesoda25** [comment]
> I'd complain "what if a sec officer gets commandphobia or a lizard is scared of snakes" but then I remembered we have psicodine if that works

either way its funny, also i dont think lizards can get lizard phobia anymore, but the way this was done might bypass that

**Review by 81Denton — CHANGES_REQUESTED** [review]
The quirk is missing remove() so that the phobia gets removed when the quirk gets removed.

**Comment by wesoda25** [comment]
Ok, I'm kinda having trouble with this (knowing it would work) since I don't know how admins would go about removing the quirk. using the path "(earliershit)/phobia/remove()" how would admins activate it?

**Comment by Putnam3145** [comment]
using the "remove quirk" verb in the view variables dropdown

**Comment by wesoda25** [comment]
Sorry it took so long i was temp git banned. Anyways a weird bug I noticed is that you don’t get the gain message for a phobia roundstart, but you do as a late arrival. I’m not sure if this is because the trauma system isn’t fully ticking yet or something, and while its not a horrific bug I’d like to sort it out. Does anyone have any ideas why it happens like this?

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
You need to have the trauma referenced on the quirk because you need to remove JUST the trauma added which is not what you're doing.

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -328,6 +328,23 @@
...
 
+/datum/quirk/phobia
+	name = "Phobia"
+	desc = "You have a debilitating fear you just can't seem to overcome."
+	value = -2
+	gain_text = null
+	lose_text = null
+	medical_record_text = "Patient has a deep rooted phobia."
+
+/datum/quirk/phobia/add()
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]: use on spawn and see if that fixes this

Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]: is there a difference if you use `on_spawn()`?

Comment (wesoda25) [reply code/datums/traits/negative.dm]: Id imagine admins wouldn’t be able to add the quirk to someone

Comment (ExcessiveUseOfCobblestone) [reply code/datums/traits/negative.dm]: nah they call `on_spawn` with their add-minabuse.

To my understanding, on_spawn should be the "default"

Comment (wesoda25) [reply code/datums/traits/negative.dm]: on spawn provides no noticeable difference

**Comment by wesoda25** [comment]
> You need to have the trauma referenced on the quirk because you need to remove JUST the trauma added which is not what you're doing.

ok ill try and figure this out

File: code/datums/traits/negative.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/datums/traits/negative.dm]: dunno why they're different anyways but 🤷‍♂

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -328,6 +328,23 @@
...
 			quirk_holder.put_in_hands(I)
 
+/datum/quirk/phobia
+	name = "Phobia"
+	desc = "You have a debilitating fear you just can't seem to overcome."
+	value = -2
+	gain_text = null
+	lose_text = null
+	medical_record_text = "Patient has a deep rooted phobia."
+
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]:
```DM
var/datum/brain_trauma/mild/phobia/the_phobia
```

File: code/datums/traits/negative.dm [hunk 3]
```diff
@@ -328,6 +328,23 @@
...
+/datum/quirk/phobia/add()
+	var/datum/brain_trauma/mild/phobia/T = new()
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]:
```DM
	the_phobia = new() //if you have to type it REEEEEE
```

File: code/datums/traits/negative.dm [hunk 4]
```diff
@@ -328,6 +328,23 @@
...
+	var/mob/living/carbon/human/H = quirk_holder
+	H.gain_trauma(T, TRAUMA_RESILIENCE_ABSOLUTE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]:
```DM
	H.gain_trauma(the_phobia, TRAUMA_RESILIENCE_ABSOLUTE)
```

File: code/datums/traits/negative.dm [hunk 5]
```diff
@@ -328,6 +328,23 @@
...
+
+/datum/quirk/phobia/remove()
+	var/mob/living/carbon/human/H = quirk_holder
+	H.cure_trauma_type(resilience = TRAUMA_RESILIENCE_ABSOLUTE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/datums/traits/negative.dm]: pretty sure this can just be `qdel(the_phobia)`, I don't see a "cure (specific) trauma" type proc.

Comment (wesoda25) [reply code/datums/traits/negative.dm]: unless i did something wrong qdel does not appear to work, but im not sure if im defining the phobia properly

Comment (ExcessiveUseOfCobblestone) [reply code/datums/traits/negative.dm]: VV the quirk and see if it refers to the phobia on the mob?

File: code/datums/traits/negative.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/datums/traits/negative.dm]: you have to type it, sad

**Comment by optimumtact** [comment]
closing as stale

**Pull request closed without merging**